tests/phpunit/tests/theme/wpThemeJson.php

```diff
@@ -3493,4 +3493,335 @@ function test_get_styles_for_block_with_content_width() {
+	 *
+	 * @ticket 56467
+	 *
+	 * Tests generating the spacing presets array based on the spacing scale provided.
+	 */
+	function test_set_spacing_sizes( $spacing_scale, $expected_output ) {
+		$theme_json = new WP_Theme_JSON(
+			array(
+				'version'  => 2,
+				'settings' => array(
+					'spacing' => array(
+						'spacingScale' => $spacing_scale,
+					),
+				),
+			)
+		);
+
+		$theme_json->set_spacing_sizes();
+		$this->assertSame( $expected_output, _wp_array_get( $theme_json->get_raw_data(), array( 'settings', 'spacing', 'spacingSizes', 'default' ) ) );
+	}
+
+	/**
+	 * Data provider for spacing scale tests.
+	 *
+	 * @return array
+	 */
+	function data_generate_spacing_scale_fixtures() {
+		return array(
+			'one_step_spacing_scale' => array(
+				'spacingScale'    => array(
+					'operator'   => '+',
+					'increment'  => 1.5,
+					'steps'      => 1,
+					'mediumStep' => 4,
+					'unit'       => 'rem',
+				),
+				'expected_output' => array(
+					array(
+						'name' => 'Medium',
+						'slug' => '50',
+						'size' => '4rem',
+					),
+				),
+			),
+
+			'two_step_spacing_scale_should_add_step_above_medium' => array(
+				'spacingScale'    => array(
+					'operator'   => '+',
+					'increment'  => 1.5,
+					'steps'      => 2,
+					'mediumStep' => 4,
+					'unit'       => 'rem',
+				),
+				'expected_output' => array(
+					array(
+						'name' => 'Medium',
+						'slug' => '50',
+						'size' => '4rem',
+					),
+					array(
+						'name' => 'Large',
+						'slug' => '60',
+						'size' => '5.5rem',
+					),
+				),
+			),
+
+			'three_step_spacing_scale_should_add_step_above_and_below_medium' => array(
+				'spacingScale'    => array(
+					'operator'   => '+',
+					'increment'  => 1.5,
+					'steps'      => 3,
+					'mediumStep' => 4,
+					'unit'       => 'rem',
+				),
+				'expected_output' => array(
+					array(
+						'name' => 'Small',
+						'slug' => '40',
+						'size' => '2.5rem',
+					),
+					array(
+						'name' => 'Medium',
+						'slug' => '50',
+						'size' => '4rem',
+					),
+					array(
+						'name' => 'Large',
+						'slug' => '60',
+						'size' => '5.5rem',
+					),
+				),
+			),
+
+			'even_step_spacing_scale_steps_should_add_extra_step_above_medium' => array(
+				'spacingScale'    => array(
+					'operator'   => '+',
+					'increment'  => 1.5,
+					'steps'      => 4,
+					'mediumStep' => 4,
+					'unit'       => 'rem',
+				),
+				'expected_output' => array(
+					array(
+						'name' => 'Small',
+						'slug' => '40',
+						'size' => '2.5rem',
+					),
+					array(
+						'name' => 'Medium',
+						'slug' => '50',
+						'size' => '4rem',
+					),
+					array(
+						'name' => 'Large',
+						'slug' => '60',
+						'size' => '5.5rem',
+					),
+					array(
+						'name' => 'X-Large',
+						'slug' => '70',
+						'size' => '7rem',
+					),
+				),
+			),
+
+			'if_bottom_end_will_go_below_zero_should_add_extra_steps_above_medium_instead' => array(
```

Comment:
Dataset names should be written with this in mind:

```
<test method name> with <dataset name>
```
In this instance (and without any suggestions in this review implemented), the output with the `--testdox` flag would be:
```
set spacing sizes with if_bottom_end_will_go_below_zero_should_add_extra_steps_above_medium_instead
```
So, while we're not writing an instruction manual on grammar, we _are_ telling a story of the expected behaviour. IMO, it's worth making this easier to read and understand what's being tested, and the expected outcome.

Also, dataset names don't need to have an `_` between words and can just be written as `the dataset name` for readability in source and CLI.

🔢 Applies to other instances in the PR.

Reply:
Thanks for the explanation, I have reworded all of these in a way which hopefully makes more sense